### DISCUSSION TITLE
Dynamic Plugins in degraded state since  4.14.0-0.okd-2024-01-06-084517 update

### DISCUSSION THREAD

**Discussion by jbparfum** [okd-project/okd#1854] [Q&A]
After upgrading to  4.14.0-0.okd-2024-01-06-084517 the dynamic plugins are in degraded state
mce
acm
monitoring-plugin

Disable/enable didn't help

Any ideas?

**Comment by melledouwsma** [comment]
Please see https://github.com/okd-project/okd/issues/1852#issuecomment-1884827211.

**Comment by vrutkovs** [comment]
Also known issue - https://issues.redhat.com/browse/OCPBUGS-26983

**Comment by laralar** [comment]
Same issue with 4.17.0-0.okd-scos-2024-11-07-025119

I noticed 

Failed to get a valid plugin manifest from /api/plugins/kubevirt-plugin/
r: failed to send GET request for "kubevirt-plugin" plugin: Get "https://kubevirt-console-plugin-service.kubevirt-hyperconverged.svc.cluster.local:9443/plugin-manifest.json": EOF

Failed to get a valid plugin manifest from /api/plugins/monitoring-plugin/
r: failed to send GET request for "monitoring-plugin" plugin: Get "https://monitoring-plugin.openshift-monitoring.svc.cluster.local:9443/plugin-manifest.json": EOF

Failed to get a valid plugin manifest from /api/plugins/networking-console-plugin/
r: failed to send GET request for "networking-console-plugin" plugin: Get "https://networking-console-plugin.openshift-network-console.svc.cluster.local:9443/plugin-manifest.json": EOF

In the following advisor I saw:
https://access.redhat.com/solutions/7049163



> Starting with Red Hat OpenShift Container Platform 4.14, the monitoring pages in the Observe section of the Red Hat OpenShift Container Platform web console are deployed as a dynamic plugin. With this change, the Cluster Monitoring Operator (CMO) is now the component that deploys the Red Hat OpenShift Container Platform web console monitoring plugin resources in the openshift-monitoringnamespace.
> Customers applying NetworkPolicy in namespace such as openshift-monitoring (which Red Hat does not recommend doing), are advised to adjust the NetworkPolicy in openshift-monitoring to allow traffic from openshift-console namespace for the service called monitoring-plugin.openshift-monitoring.svc.cluster.local on port 9443.

**Comment by glennodickson** [comment]
SImilar issue here with 4.18. and have noted [here ](https://github.com/okd-project/okd/issues/2191) for any resolution or suggestions.

**Reply by laralar** [reply]
My particular case was a DNS issue.  My dns was including search everything as search domains.  That was taking priority over fqdn in kubernetes.  The search domains should only be for *apps... 

https://www.reddit.com/r/openshift/comments/1bdpion/comment/m1w2o31/?utm_source=share&utm_medium=mweb3x&utm_name=mweb3xcss&utm_term=1&utm_content=share_button

**Comment by glennodickson** [comment]
What corrective action did you apply ?

**Comment by laralar** [comment]
As i mentioned. In my case i had too many search domains. I don't know if it's your case. It had to do with the DNS configuration. I was using another DNS service and there were too mmany search domains being assigned by the DHCP .  It had nothing to do with the okd installation.

The DHCP should not give any search domains
 

Remove all search domains from your DNS server. There should be no *.domainname


For example of your domain name is 

okd.example.com and DHCP provides IP addresses in that domain.  Okd.example.com should not have any @ entry. My DNS was propagating okd.example.com as search domain. Therefore that was taking precedence over the plugin fqnd

Configure a DNS in a VM exactly as the documentation instructions. And use only that DNS server.

**Comment by glennodickson** [comment]
Hi
Thanks heaps for your description. I've previously installed and run a
cluster for sometime with version 4+. After upgrading to 4.17 I've had a
plugin issue where the console would come up however, without the network
and Observe menu items. Then upgraded to 4.18 and found the console unable
to come up at all in the browser.
All using the exact same configuration for DNS etc as I use a script to
create for consistency.

it certainly is due to the NDOT=5 and appending the domain name to make a
FQDN when resolving the monitoring and network service. I'm not sure why it
is having a problem with only those services and not the entire cluster?
Since I use a script to create and configure DNS then it is the 4.14
dynamic plugin change of version 4.14+ which is the only thing which has
changed.

Any ideas would be appreciated.
Regards
Glenn
